EpiJS/plots.js

```diff
@@ -7,102 +7,119 @@
     }
 
-    let sirChart = new Chart(canvas, {// skipcq: JS-0502
-        type: 'line',
-        data: data,
-        options: options 
-    });
-    return sirChart;
-}
+    // For every day until `time`, get the data
+    for (let x = 0; x < time; x++) {
+        let current = model.get_data(x+1)
 
-/**
- * Manipulate the chart.js graph
- * 
- * @param id The chart.js graph
- * @param {String} mvalue The value to manipulate in `chart.data.datasets[x]`. This can be any valid chart.js parameter. See https://www.chartjs.org/docs/latest/charts/line.html#line-styling
- * @param value The value to insert into the graph
- * @returns The chart.js graph
- * 
- * @example
- * 
- * let sirout1 = sir(4, 9999, 1000, 100, 1/21, 10999, true)
- *
- * let sirplot = plot(sirout1, "canvas-pre1", 100)
- * 
- * sirplot.manipulate(sirplot, "fill", true) // Set fill to true
- */
-function manipulate(id, mvalue, value) {
-    var manip = Chart.getChart(id)// skipcq: JS-0502
+        // For every compartment, get the data
+        for (let y in model.compartments) {
+            // If the compartment data is an object, get .population
+            if (typeof current[model.compartments[y][1]] === 'object') {
+                data.yvals[model.compartments[y][1]].push(current[model.compartments[y][1]].population)
+            } else {
+                data.yvals[model.compartments[y][1]].push(current[model.compartments[y][1]])
+            }
+        }
+        data.xvals.push(x+1)
+    }
 
-    for (var x in manip.data.datasets) {
-        manip.data.datasets[x][mvalue] = value
+    // Create the plot
+    let plotlyData = []
+    for (let x in data.yvals) {
+        plotlyData.push({
+            x: data.xvals,
+            y: data.yvals[x],
+            name: x,
+            mode: 'lines'
+        })
     }
 
-    manip.update()
-    return manip
+    if (typeof process === 'object' && String(process) === '[object process]') { // Is running in NodeJS
+        // Create http server
+        let server = http.createServer((req, res) => {
+            // Send the plot as HTML
+            res.writeHead(200, {'Content-Type': 'text/html'})
+            res.write(`<!DOCTYPE html>
+            <html>
+            <head>
+            <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
+            </head>
+            <body>
+            <div id="plot"></div>
+            <script>
+            let data = ${JSON.stringify(plotlyData)}
+            let layout = {
+                title: '${title}',
+                xaxis: {
+                    title: 'Days'
+                },
+                yaxis: {
+                    title: 'Cases'
+                }
+            }
+            Plotly.newPlot('plot', data, layout)
+            </script>
+            </body>
+            </html>`)
+            res.end()
+        })
+
+        // Start the server
+        server.listen(8080)
+        console.log('Plotly server running on http://localhost:8080')
```

Comment:
This message seems a bit ambiguous; add a source in brackets to make it more relevant (e.g. `[@epispot/epijs]`)
```suggestion
        console.log('[@epispot/epijs] Plotly server running on http://localhost:8080')
```

Reply:
Also we may want to add a port argument to avoid interference with other `localhost`s